jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java

```diff
@@ -143,6 +144,15 @@ private SpotLight buildSpotLight(JsonObject obj) {
             outerConeAngle = FastMath.HALF_PI - 0.000001f;
         }
 
+        if(APPLY_INTENSITY_CONVERSION)  {
+            float solidAngle = 2.0f * FastMath.PI * (1.0f - FastMath.cos(outerConeAngle));
+            intensity = intensity / solidAngle;
+        }
+
+        float range = obj.has("range") ? obj.get("range").getAsFloat() : (COMPUTE_LIGHT_RANGE ? getCutoffDistance(color, intensity) : Float.POSITIVE_INFINITY);
```

Comment:
The POSITIVE_INFINITY range was an error to begin which and required a patch to turn it into FLOAT_MAX_VALUE https://github.com/jMonkeyEngine/jmonkeyengine/pull/2058 , but ends up introducing some floating point errors with large numbers that end up making the lighting calculation a bit off.

Truth is: jme doesn't support infinite lights very well, so instead, this patch calculates the right range based on the falloff function.